hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/types/HazelcastObjectType.java

```diff
@@ -36,16 +37,16 @@ public class HazelcastObjectType extends RelDataTypeImpl {
     private final String name;
     private final boolean nullable;
     /** Modifiable list of fields to support cyclic types. */
-    private List<Field> fields = new ArrayList<>();
+    private List<Field> fields = new NonTraversableList<>("Type fields must be finalized");
     /** Cached list of field names. */
     private List<String> fieldNames;
 
-    /** Not usable until {@link #finalizeFields()} is called. */
+    /** Not usable until {@link #finalizeFields} is called. */
```

Comment:
this comment while technically true, is a bit of a shortcut. You do not not call `finalizeFields` directly on this object, but on a map containing it.

Reply:
https://github.com/hazelcast/hazelcast/pull/26056#discussion_r1419008654

Reply:
```suggestion
    /** Not usable until {@link #finalizeFields} with map containing this type is called. */
```
?

Reply:
c1fe5fbd